Python/ceval.c

```diff
@@ -5697,10 +5771,7 @@ _PyEvalFrameClearAndPop(PyThreadState *tstate, InterpreterFrame * frame)
 {
     --tstate->recursion_remaining;
     assert(frame->frame_obj == NULL || frame->frame_obj->f_own_locals_memory == 0);
-    if (_PyFrame_Clear(frame, 0)) {
-        ++tstate->recursion_remaining;
-        return -1;
-    }
+    _PyFrame_Clear(frame, 0);
```

Comment:
Why aren't we checking this for failure anymore? Because of bpo-45786? It still looks to me like the exiting frame could return a non-`NULL` value with a `MemoryError` set, though...

If it really doesn't matter, perhaps make both `_PyFrame_Clear` and `_PyEvalFrameClearAndPop` return `void`?

Reply:
I think we are in trouble if that happens anyway, so we need to fix bpo-45786.
Making `_PyFrame_Clear` return void for now makes sense.